lib/ash/type/type.ex

```diff
@@ -50,9 +54,13 @@ defmodule Ash.Type do
 
   ## Built in types
 
-  #{Enum.map_join(@short_names, fn {key, module} -> "* `#{inspect(key)}` - `#{inspect(module)}`\n" end)}
+  #{Enum.map_join(@builtin_short_names, fn {key, module} -> "* `#{inspect(key)}` - `#{inspect(module)}`\n" end)}
+
+  #{if @custom_short_names != [], do: "## Custom types"}
```

Comment:
Honestly this is probably unnecessary. When generating hex docs for Ash, it will always be empty, whereas when generating hex docs in user's applications they won't be generating it for dependencies. I think we can remove this.